.github/workflows/docs.yml

```diff
@@ -0,0 +1,53 @@
+# https://github.com/mitmproxy/pdoc/blob/main/.github/workflows/docs.yml
+
+name: PyHa Documentation
+
+# build the documentation whenever there are new commits on main
+on:
+  push:
+    branches:
+      - main
+      - np-generate-docs
+
+# security: restrict permissions for CI jobs.
+permissions:
+  contents: read
+
+jobs:
+  # Build the documentation and upload the static HTML files as an artifact.
+  build:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip poetry
+          poetry install
+
+      - name: Build Documentation
+        run: poetry run python -m pdoc ./PyHa --docformat google -o docs/
+        env:
+          PDOC_ALLOW_EXEC: 1
+
+      - uses: actions/upload-pages-artifact@v3
+        with:
+          path: docs/
+
+  # Deploy the artifact to GitHub pages.
+  # This is a separate job so that only actions/deploy-pages has the necessary permissions.
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+    permissions:
+      pages: write
+      id-token: write
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    steps:
+      - id: deployment
+        uses: actions/deploy-pages@v4
```

Comment:
If we allow `pdoc` to run on all branches, this job should get a `if` clause to only execute on the `main` branch.